### PR TITLE
(#19967): Fixing identifier deletion when child contentlet present

### LINKED ISSUE #19967: Create a Host Integrity Checker
If we move to the idea of a hostName acting as the site key for various functionality, such as apps, push publishing and to calculate known ids based on asset paths, we will need to be able to check if the hostname/host.identifiers match between two environments.  In order to do this we should have an integrity checker than verifies that the hostNames+ids match and if not, allow an administrator to fix either locally or remotely.

### PR DESCRIPTION
**Pull request by victoralfaro-dotcms**
The introduced changes fix the fact that contentlet has children and the host integrity checker will try to fix conflicts.

### CODE REVIEW

**Review by dsilvam — APPROVED** [review]

**Review by nollymar — COMMENTED** [review]
Can a test be implemented?

File: dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java [hunk 1]
```diff
@@ -75,8 +74,7 @@ public File generateCSVFile(final String outputPath) throws DotDataException, IO
             writer = new CsvWriter(new FileWriter(csvFile, true), '|');
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java:74]: Double check if the db annotations are in place

File: dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java [hunk 2]
```diff
@@ -325,13 +351,19 @@ private void fixHostConflicts(final Map<String, Object> row,
...
         if (results == null || results.isEmpty()) {
-            final String temporalAssetName = "TEMP_" + UUID.randomUUID();
-            dc.setSQL("INSERT INTO identifier (id, parent_path, asset_name, host_inode, asset_type, syspublish_date, sysexpire_date)" +
-                    " SELECT ?, parent_path, '" + temporalAssetName + "', host_inode, asset_type, syspublish_date, sysexpire_date" +
-                    " FROM identifier WHERE id = ?");
-            dc.addParam(remoteHostIdentifier);
-            dc.addParam(localHostIdentifier);
-            dc.loadResult();
+            Logger.debug(
+                    this,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java:355]: use the supplier log

Logger.debug(this, ()-> "Hi Viko")

File: dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java [hunk 3]
```diff
@@ -579,7 +576,23 @@ private void insertHostForInode(final String oldHostIdentifier,
...
+                " FROM contentlet c INNER JOIN contentlet_version_info cvi on (c.inode = %s AND c.language_id = cvi.lang) WHERE c.identifier = ? and c.language_id = ?",
+                cviInodeColumn);
+        final String finalSql = DbConnectionFactory.isMySql() ? insertHostSql.replaceAll("\"", "`") : insertHostSql;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java:594]: keep in mind MySQL is not longer used

**Review by jdotcms — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java (hunk 2 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java:355]: Done.

File: dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java (hunk 3 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java:594]: Removed.

File: dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java (hunk 1 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java:76]: Checked and added.

File: dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java (hunk 3 above)
Comment (victoralfaro-dotcms) [reply dotCMS/src/main/java/com/dotcms/integritycheckers/HostIntegrityChecker.java]: Removed.

**Review by nollymar — APPROVED** [review]

**Pull request merged**